src/transformStateWithClones.js

```diff
@@ -1,13 +1,41 @@
 'use strict';
 
+const ADD_PROPERTIES = 'addProperties';
+const REMOVE_PROPERTIES = 'removeProperties';
+const CLEAR = 'clear';
+
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateLog = [];
+
+  for (const act of actions) {
```

Comment:
```suggestion
  for (const action of actions) {
```
would be more reasonable, but that also works 😉